gk/fib.c

```diff
@@ -60,45 +65,113 @@ gk_lpm_add_ipv6_route(uint8_t *ip,
  * callers must ensure that the entry is not being used.
  */
 static inline int
-initialize_ether_cache(struct ether_cache *eth_cache)
+clear_ether_cache(struct ether_cache *eth_cache)
 {
-	memset(eth_cache, 0, sizeof(*eth_cache));
+	memset(eth_cache->fields_to_clear, 0, sizeof(*eth_cache) -
+		offsetof(struct ether_cache, fields_to_clear));
+
+	if (rte_atomic32_read(&eth_cache->ref_cnt) != 0) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"gk: the value of ref_cnt field in Ethernet cache entry is %d rather than 0 while calling function %s!\n",
+			rte_atomic32_read(&eth_cache->ref_cnt), __func__);
```

Comment:
Save the value of the first call to `rte_atomic32_read()` since the bug could be so bizarre that the second call returns zero!